### PR TITLE
Fix #261: port local repository workflows to native Windows

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

- resolve local Git and GitHub CLI executables explicitly, including native Windows `PATH`/`PATHEXT` behavior and `JEFE_GIT_BIN` / `JEFE_GH_BIN` overrides
- keep every Git and `gh` value as a distinct argv item, preserving spaces, Unicode, refs, prompts, and paths without a local shell
- add Windows-aware local-path comparison while preserving original display values, support long local paths, and reject UNC work directories early with actionable guidance
- replace broad local `git clean` / `reset --hard` cleanup with enumerated untracked-file deletion and per-path tracked restoration, preserving `.jefe` and `.llxprt` metadata and surfacing Windows file-lock failures
- retain the existing remote Linux command planner unchanged
- add real-Git tests under paths with spaces and Unicode, adversarial filenames, owned metadata preservation, path-policy tests, executable-resolution tests, and a deterministic Windows local-preparation TUI scenario

## Test-first evidence

- RED: path-policy tests initially failed because the comparison/validation API did not exist
- RED: executable-resolution tests initially failed because the typed local-tool resolver did not exist
- GREEN: focused real-Git cleanup, resolver, and path-policy suites pass
- REFACTOR: cleanup was isolated into its own filesystem-boundary module and command resolution was shared by Git, GitHub operations, and GitHub auth

## Verification

- `cargo fmt --all --check`
- `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- `cargo build --workspace --all-features --locked`
- `cargo test --workspace --all-features --locked`

All passed on native Windows.

Fixes #261

### CODE REVIEW

File: src/app_input/issue_cleanup.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            "-z",
+        ],
+    )?;
+    require_success(&output, "ls-files --others --exclude-standard -z")?;
+    let untracked = parse_changed_paths(&output.stdout)?;
+    for relative in &untracked {
+        remove_untracked_path(work_dir, relative)?;
+    }
+    remove_empty_untracked_parents(work_dir, &untracked);
+    restore_tracked_paths(work_dir)
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:31]: Verified against current behavior. This ordering is deliberate: untracked deletion is the irreversible/high-risk step, so tracked restoration must not mutate user files if an untracked deletion fails. Reversing the order creates that exact unsafe partial mutation. Rollback would require snapshotting arbitrary user content and is outside this cleanup contract.

File: src/app_input/issue_cleanup.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        work_dir,
+        [
+            "-c",
+            "core.quotepath=false",
+            "ls-files",
+            "--others",
+            "--exclude-standard",
+            "-z",
+        ],
+    )?;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:24]: No change: Git does not track empty directories, so a genuinely empty directory does not make a worktree dirty and is outside Git dirty-worktree cleanup semantics. Parent directories of enumerated untracked files are already removed when empty.

File: src/app_input/issue_cleanup.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        std::fs::remove_dir(&path)
+    } else {
+        std::fs::remove_file(&path)
+    };
+    result.map_err(|error| {
+        format!(
+            "Failed to remove untracked path {}: {error}. Close programs using the file and retry.",
+            path.display()
+        )
+    })
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:128]: Acknowledged as a diagnostic enhancement, not a correctness blocker. The message includes the actual OS error (including permission/read-only details) plus the common Windows lock remedy.

File: src/app_input/issue_cleanup.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            [
+                std::ffi::OsStr::new("restore"),
+                std::ffi::OsStr::new("--source=HEAD"),
+                std::ffi::OsStr::new("--staged"),
+                std::ffi::OsStr::new("--worktree"),
+                std::ffi::OsStr::new("--"),
+                relative.as_os_str(),
+            ],
+        )?;
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:59]: Keeping per-path restore intentionally. Paths can be arbitrarily long and non-UTF-8 on Unix; individual invocations avoid Windows command-line limits and preserve exact path arguments. Correctness and bounded argv size take priority over process-count optimization here.

File: src/app_input/issue_cleanup.rs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            work_dir,
+            [
+                std::ffi::OsStr::new("restore"),
+                std::ffi::OsStr::new("--source=HEAD"),
+                std::ffi::OsStr::new("--staged"),
+                std::ffi::OsStr::new("--worktree"),
+                std::ffi::OsStr::new("--"),
+                relative.as_os_str(),
+            ],
+        )?;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:58]: Keeping per-path restore intentionally to guarantee command-line-length safety and exact non-UTF-8 path handling. A single invocation reintroduces Windows argv-limit failures.

File: src/local_command.rs [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    #[cfg(unix)]
+    if platform == ToolPlatform::Unix {
+        use std::os::unix::fs::PermissionsExt;
+        return path
+            .metadata()
+            .is_ok_and(|metadata| metadata.permissions().mode() & 0o111 != 0);
+    }
+    let _ = platform;
+    true
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/local_command.rs:161]: No change: following PATH symlinks is standard executable lookup behavior on Unix (for example /usr/bin tools commonly resolve through links). Rejecting symlinks would break normal installations without providing a meaningful boundary because PATH itself is trusted configuration.

File: src/app_input/issue_cleanup.rs [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:59]: Keeping per-path restore intentionally: this guarantees OS argv-limit safety for arbitrary path sets and preserves exact path values. The process overhead is an accepted tradeoff for reliable cleanup.

File: src/app_input/issue_cleanup.rs [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        ],
+    )?;
+    require_success(&output, "ls-files --others --exclude-standard -z")?;
+    let untracked = parse_changed_paths(&output.stdout)?;
+    for relative in &untracked {
+        remove_untracked_path(work_dir, relative)?;
+    }
+    remove_empty_untracked_parents(work_dir, &untracked)?;
+    restore_tracked_paths(work_dir)
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:32]: No change after verification. Deletion-first is intentional so failed untracked deletion cannot be followed by tracked-file mutation; rollback of arbitrary user content is not safely possible without a separate snapshot system. Empty directories do not affect Git dirty state.

File: src/app_input/issue_cleanup.rs [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+                std::ffi::OsStr::new("--source=HEAD"),
+                std::ffi::OsStr::new("--staged"),
+                std::ffi::OsStr::new("--worktree"),
+                std::ffi::OsStr::new("--"),
+                relative.as_os_str(),
+            ],
+        )?;
+    }
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:61]: Keeping bounded per-file restore intentionally. It avoids Windows argv limits for arbitrary path lengths and preserves exact pathspecs; correctness outweighs process-spawn optimization in this user-confirmed cleanup path.

File: src/app_input/issue_cleanup.rs [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    } else {
+        std::fs::remove_file(&path)
+    };
+    result.map_err(|error| {
+        format!(
+            "Failed to remove untracked path {}: {error}. Close programs using the file and retry.",
+            path.display()
+        )
+    })
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:129]: Verified the current Rust/Windows behavior and did not apply this suggestion: symlink_metadata reports directory symlinks as symlinks, and removal behavior differs by toolchain/Windows API evolution. The proposed metadata.is_dir premise is not reliable for junction classification; using recursive removal would create target-traversal risk. This needs a dedicated typed reparse-point boundary rather than a speculative change.

File: src/app_input/issue_cleanup.rs [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+use std::path::{Path, PathBuf};
+
+#[cfg(unix)]
+use std::ffi::OsString;
+#[cfg(unix)]
+use std::os::unix::ffi::OsStringExt;
+
+use super::{git_capture, git_require_success, require_success};
+
+pub(in crate::app_input) fn discard_workdir_changes(work_dir: &Path) -> Result<(), String> {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:13]: Acknowledged. This is an explicitly user-confirmed destructive operation; Jefe does not claim an exclusive repository lock. Every removal is ownership-filtered and inspected immediately before deletion, while Git performs tracked restoration. Adding cross-process repository locking is outside this PR.

[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:13]: Acknowledged as a broader API design improvement. The existing preparation boundary is Result<_, String> throughout and the strings preserve the concrete OS/Git diagnostics. Introducing a new typed error hierarchy would be a larger cross-module change beyond #261.

File: src/app_input/issue_git_prep.rs [hunk 12]
```diff
@@ -15,15 +15,18 @@
...
+#[path = "issue_cleanup.rs"]
+mod issue_cleanup;
 /// Safety guards for the destructive force-reclone path (issue #190).
 ///
 /// Re-exported here so callers can reference
 /// `super::issue_git_prep::validate_reclone_target` regardless of the internal
 /// module split.
 #[path = "reclone_safety.rs"]
 mod reclone_safety;
+pub(super) use issue_cleanup::discard_workdir_changes;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_git_prep.rs:29]: The semantic change is intentional for #261: broad git clean/reset were explicitly removed to protect .jefe/.llxprt and use ownership-scoped native operations. Git does not track standalone empty directories, so they do not affect dirty status. Per-path restore is intentionally argv-bounded and prevents broad destructive reset behavior.

File: src/app_input/issue_git_prep.rs [hunk 13]
```diff
@@ -524,72 +527,15 @@ pub(super) fn prepare_issue_workdir(work_dir: &Path) -> PrepResult {
...
-    // Now discard tracked modifications. Running after clean so if clean
-    // fails, tracked changes are still intact.
-    git_require_success(work_dir, ["reset", "--hard"])?;
-    Ok(())
-}
-
 /// Run `git` with the given args in `work_dir`, capturing output. Returns an
 /// error string on spawn failure (does not inspect exit status).
-fn git_capture<I, S>(work_dir: &Path, args: I) -> Result<std::process::Output, String>
+pub(super) fn git_capture<I, S>(work_dir: &Path, args: I) -> Result<std::process::Output, String>
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_git_prep.rs:532]: No change. The paired OCR review correctly notes pub(super) is the narrow visibility required by the split module arrangement and existing caller/test seams. It remains module-scoped, not crate-public.

File: src/app_input/issue_git_prep.rs [hunk 14]
```diff
@@ -637,7 +583,7 @@ where
 
 /// Inspect a captured `git` output and return `Err` with stderr/stdout detail
 /// when the exit status was non-zero.
-fn require_success(output: &std::process::Output, context: &str) -> Result<(), String> {
+pub(super) fn require_success(output: &std::process::Output, context: &str) -> Result<(), String> {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_git_prep.rs:586]: No change. As with git_capture, pub(super) is the established narrow visibility needed by the split helper module and parent test seam; it is not exposed outside app_input.

File: src/services/normalize.rs [hunk 15]
```diff
@@ -4,6 +4,114 @@
...
+    left: &str,
+    right: &str,
+    platform: LocalPathPlatform,
+) -> bool {
+    normalize_local_path(left, platform) == normalize_local_path(right, platform)
+}
+
+fn normalize_local_path(value: &str, platform: LocalPathPlatform) -> String {
+    let separated = match platform {
+        LocalPathPlatform::Windows => value.replace('\\', "/").to_lowercase(),
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/services/normalize.rs:51]: No change. Unicode lowercasing is intentional because Windows path matching is Unicode case-insensitive in ordinary user-facing paths; reverting to ASCII-only comparison would reintroduce the tested Müller/müller mismatch. Exact NT ordinal folding would require a Windows-specific API and is outside this normalization helper.

[the remark this answers is not in the document]
Comment (acoliver) [reply src/services/normalize.rs:51]: No change: this path normalization runs only during repository form/preparation operations, not a hot loop. Clarity and tested correctness outweigh eliminating one short-lived allocation.

File: src/app_input/issue_cleanup.rs [hunk 16]
```diff
@@ -0,0 +1,185 @@
...
+                std::ffi::OsStr::new("restore"),
+                std::ffi::OsStr::new("--source=HEAD"),
+                std::ffi::OsStr::new("--staged"),
+                std::ffi::OsStr::new("--worktree"),
+                std::ffi::OsStr::new("--"),
+                relative.as_os_str(),
+            ],
+        )?;
+    }
+    Ok(())
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:60]: Keeping per-path restore intentionally to bound argv size on Windows and preserve arbitrary exact pathspecs. The performance tradeoff is preferable to command-line-length failures in a user-confirmed cleanup path.

[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_cleanup.rs:60]: Duplicate of the batching suggestion. Per-path restore is intentional for Windows argv-length safety and exact path handling.

File: src/state/issues_tests_send_to_agent.rs [hunk 17]
```diff
@@ -42,7 +42,6 @@ fn issue_detail() -> IssueDetail {
         comments: Vec::new(),
         has_more_comments: false,
         comments_cursor: None,
-        issue_type_name: None,
     }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests_send_to_agent.rs:45]: The field remains initialized once at line 39. This commit removed only the duplicate initializer introduced when current main added the same field to the fixture; the full struct literal compiles in all CI jobs.

File: src/local_command.rs [hunk 18]
```diff
@@ -0,0 +1,240 @@
...
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/local_command.rs:161]: No change. On Windows, PATHEXT defines which extensions are treated as executable; candidate generation already restricts lookup to exactly those configured extensions. If a user explicitly adds .TXT to PATHEXT, Windows shell semantics intentionally make that extension executable. A second extension check would duplicate the same policy without adding safety.

File: src/state/issues_tests_send_to_agent.rs (hunk 17 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply src/state/issues_tests_send_to_agent.rs:45]: False positive: the helper contains issue_type_name: None exactly once. The reviewed diff removed a duplicate initializer introduced by merge overlap. Current head compiles and all Build/Test/Windows jobs pass.

File: src/app_input/issue_git_prep.rs (hunk 13 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_git_prep.rs:532]: False positive: Rust child modules can access private items in ancestor modules, and pub(super) is at least as permissive for this child import. issue_cleanup compiles with use super::git_capture; all platform builds pass. Broadening to pub(crate) would unnecessarily expand visibility.

File: src/app_input/issue_git_prep.rs (hunk 14 above)
[the remark this answers is not in the document]
Comment (acoliver) [reply src/app_input/issue_git_prep.rs:586]: Duplicate false positive. Child modules can access ancestor items; pub(super) does not prevent issue_cleanup from importing require_success. Current head compiles on Linux and Windows, so no visibility expansion is needed.
